scripts/pylib/twister/twisterlib.py

```diff
@@ -771,6 +771,8 @@ def device_is_available(self, instance):
         device = instance.platform.name
         fixture = instance.testcase.harness_config.get("fixture")
         for d in self.suite.duts:
+            if not d.platform == device:
+                continue
             if fixture and fixture not in d.fixtures:
                 continue
             if d.platform != device or (d.serial is None and d.serial_pty is None):
```

Comment:
Just below is `if d.platform != device or (d.serial is None and d.serial_pty is None): continue`. Added checks looks like redundancy.